storage.lua

```diff
@@ -246,6 +247,54 @@ function mail.extractMaillists(receivers_string, maillists_owner)
 	return receivers
 end
 
+function mail.get_setting_default_value(setting_name)
+	local default_values = {
+		chat_notifications = true,
+		onjoin_notifications = true,
+		hud_notifications = true,
+		unreadcolorenable = true,
+		cccolorenable = true,
+		defaultsortfield = 3,
+		defaultsortdirection = 1,
+	}
+	return default_values[setting_name]
+end
+
+function mail.get_setting(playername, setting_name)
+	local entry = mail.get_storage_entry(playername)
+	local setting = nil
+	for _, existing_setting in ipairs(entry.settings) do
```

Comment:
Why don't you use a simple map table here instead of a list?
Iterating and appending key-value pairs is pretty error-prone and inefficient IMHO.

Reply:
> You mean ?
```lua
{
  setting_name = "setting value",
  other_setting_name = "other setting value",
}
```
instead of
```lua
{
  { name = "setting_name", value = "setting value" },
  { name = "other_setting_name", value = "other setting value"},
}
```